### PR TITLE
Hierarchy level creation error handling

### PR DESCRIPTION
**Pull request by jasdom**
Closes [472](https://github.com/iTwin/presentation/issues/472)

Hierarchy level creation errors are now displayed in tree component using `TreeModelInfoNode`.

### CODE REVIEW

File: packages/core-interop/src/test/Metadata.test.ts [hunk 1]
```diff
@@ -63,6 +63,39 @@ describe("createMetadataProvider", () => {
...
+          .stub<[SchemaKey], CoreSchema>()
+          .withArgs(matchSchemaName)
+          .resolves({
+            name: "y",
+          } as unknown as CoreSchema),
+      } as unknown as SchemaContext;
+
+      const provider = createMetadataProvider(schemaContext);
+      await provider.getSchema("x");
+      await provider.getSchema("x");
```
Comment (grigasp) [inline packages/core-interop/src/test/Metadata.test.ts:79]: I'd say this should be `await Promise.all([provider.getSchema("x"), provider.getSchema("x")])`

File: packages/hierarchies-react/src/presentation-hierarchies-react/internal/TreeLoader.ts [hunk 2]
```diff
@@ -46,25 +48,29 @@ export class HierarchyLoader implements IHierarchyLoader {
...
-            parentId: parent.id,
-            type: "ResultSetTooLarge",
-            message: err.message,
-          };
+          node.type = "ResultSetTooLarge";
+          node.message = err.message;
           return of([node]);
         }
-        throw err;
+        return of([node]);
```
Comment (grigasp) [inline packages/hierarchies-react/src/presentation-hierarchies-react/internal/TreeLoader.ts:73]: Creating an object and overwriting its attributes right away doesn't make much sense, IMO. If you don't want to repeat the same attributes, I suggest doing the following:
```ts
const nodeProps = {
  id: ...,,
  parentId: ...,
};
if (x) {
  return of([{ ...nodeProps, type: "", message: "" }]);
}
return of([{ ...nodeProsp, type: "", message: "" }]);
```

File: packages/hierarchies/src/hierarchies/HierarchyProvider.ts [hunk 3]
```diff
@@ -314,7 +316,7 @@ export class HierarchyProvider {
     const loggingCategory = `${LOGGING_NAMESPACE}.GetCachedObservableEntry`;
     const { parentNode, ...restProps } = props;
     const cached = this._nodesCache.get(props);
-    if (cached) {
+    if (!props.ignoreCache && cached) {
```
Comment (grigasp) [inline packages/hierarchies/src/hierarchies/HierarchyProvider.ts:319]: Why even request the cache entry if we're going to ignore it?

File: packages/hierarchies-react/src/presentation-hierarchies-react/internal/TreeLoader.ts [hunk 4]
```diff
@@ -59,18 +59,14 @@ export class HierarchyLoader implements IHierarchyLoader {
...
           id: `${parent.id ?? ""}-${err.message}`,
           parentId: parent.id,
-          type: "Unknown",
-          message: "Failed to create hierarchy level",
         };
         if (err instanceof RowsLimitExceededError) {
-          node.type = "ResultSetTooLarge";
-          node.message = err.message;
-          return of([node]);
+          return of([{ ...nodeProps, type: "ResultSetTooLarge", message: err.message } as TreeModelInfoNode]);
```
Comment (grigasp) [inline packages/hierarchies-react/src/presentation-hierarchies-react/internal/TreeLoader.ts:67]: why do we need `as TreeModelInfoNode`? would `type: "ResultSetTooLarge" as const,` drop this requirement?

Comment (jasdom) [reply packages/hierarchies-react/src/presentation-hierarchies-react/internal/TreeLoader.ts]: Yes, that worked, thanks.

**Review by grigasp — COMMENTED** [review]
`hierarchies` and `core-interop` look good to me, leaving `hierarchies-react` for @saskliutas

**Review by saskliutas — APPROVED** [review]
`hierarchies-react` looks good

**Pull request merged**